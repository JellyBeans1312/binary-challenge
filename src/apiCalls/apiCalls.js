export const getEvents = async () => {
  try {
    const options = { 
      method: 'GET',
      headers: {
        Authorization: 'Bearer qJhnqW5Rsc36m6cUYXZyCaV2BaFNB0QoGpr4fmk_',
        Accept: 'application/json',
        'Content-Type': 'application/json'
      }
    }

    const response = await fetch('https://api.predicthq.com/v1/events/', options)
    const result = await response.json()
    return result.results
  } catch(error) {
    console.log(error);
  }
}

export const searchEvent = async (searchParam) => {
  try {
    const options = {
      method: 'GET',
      headers: {
        Authorization: 'Bearer qJhnqW5Rsc36m6cUYXZyCaV2BaFNB0QoGpr4fmk_',
        Accept: 'application/json',
        'Content-Type': 'application/json'
      }
    }
    
    const response = await fetch(`https://api.predicthq.com/v1/events/?q=${searchParam}/`, options)
    const result = await response.json()
    return result.results

  } catch(error) {
    throw new Error(error)
  }
}

export const getAddress = async (lat, long) => {
  try {
    const response = await fetch(`https://maps.googleapis.com/maps/api/geocode/json?latlng=${lat},${long}&key=AIzaSyDPPlNyjATG8sQnfsdg45ln7kUoyFH24m0`)
    const result = await response.json();
    return result.results.map(result => {
      return result.formatted_address
    });
  } catch(error) {
    console.log(error)
  }
}