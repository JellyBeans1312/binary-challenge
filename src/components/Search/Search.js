import React, { Component } from 'react';
import { searchEvent } from '../../apiCalls/apiCalls'

export class Search extends Component {
  constructor() {
    super();
    this.state = {
      search: ''
    }
  }

  onChange = () => {

  }

  handleSubmit = async (e) => {
    e.preventDefault()
  }

  render() {
    return(
      <form onSubmit={this.handleSubmit}>
        <input
          type='text'
          placeholder='Search for an event'
          value={this.state.search}
        />
      </form>
    )
  }
}