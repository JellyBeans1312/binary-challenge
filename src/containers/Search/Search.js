import React, { Component } from 'react';
import { searchEvent } from '../../apiCalls/apiCalls'
import { connect } from 'react-redux';
import { addEvents } from '../../actions';
import PropTypes from 'prop-types';

export class Search extends Component {
  constructor() {
    super();
    this.state = {
      search: '',
      error: ''
    }
  }

  onChange = (e) => {
    this.setState({ search: e.target.value })
  }

  searchEvent = async (e) => {
    e.preventDefault()
    try {
      const searchResults = await searchEvent(this.state.search)
      this.props.addEvents(searchResults) 
    } catch(error) {
      this.setState({ error: 'There was a problem getting your search results'})
    }
  }

  render() {
    return(
      <div>
      {this.state.error}
        <form onSubmit={this.searchEvent}>
          <input
            type='text'
            placeholder='Search for an event'
            value={this.state.search}
            onChange={this.onChange}
          />
          <button>Search</button>
        </form>
      </div>
    )
  }
}

export const mapDispatchToProps = dispatch => ({
  addEvents: events => dispatch(addEvents(events))
});

export default connect(null, mapDispatchToProps)(Search)

Search.propTypes = {
  addEvents: PropTypes.func.isRequired
}