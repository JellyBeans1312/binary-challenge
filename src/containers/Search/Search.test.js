import React from 'react';
import { shallow } from 'enzyme';
import { Search } from './Search'
import { searchEvent } from "../../apiCalls/apiCalls";

jest.mock('../../apiCalls/apiCalls')

describe("Search", () => {
  let wrapper;
  let mockEvent; 

  const mockEvents = [
    { title: "something", category: 'concert', location: [{lat: 23, long: 234}] }
  ];

  beforeEach(() => {
    mockEvent = { preventDefault: jest.fn() };
    wrapper = shallow(
      <Search
        events={mockEvents}
      />
    );
    searchEvent.mockImplementation(() => {
      return Promise.resolve([{address: 'something 123 something else drive'}])
    });
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should call handleChange method when change happens", () => {
    const mockEvent = { target: { value: "earthgang" } };

    wrapper.find("input").simulate("change", mockEvent);
    expect(wrapper.state("search")).toEqual("earthgang");
  });

  it('should call searchEvent', () => {
    wrapper.instance().searchEvent(mockEvent)
    expect(searchEvent).toHaveBeenCalled()
  })

  it("sets error in state if the fetch fails", async () => {
    searchEvent.mockImplementationOnce(() => {
      throw new Error("Please Try Again");
    });

    await wrapper.instance().searchEvent(mockEvent);

    expect(wrapper.state("error")).toEqual("There was a problem getting your search results");
  });
})