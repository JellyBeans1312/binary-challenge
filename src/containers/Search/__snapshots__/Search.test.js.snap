// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should match snapshot 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Search for an event"
      type="text"
      value=""
    />
    <button>
      Search
    </button>
  </form>
</div>
`;
